src/test/java/org/opensearch/knn/index/LuceneEngineIT.java

```diff
@@ -43,6 +46,15 @@ public class LuceneEngineIT extends KNNRestTestCase {
 
     private static final float[][] TEST_QUERY_VECTORS = { { 1.0f, 1.0f, 1.0f }, { 2.0f, 2.0f, 2.0f }, { 3.0f, 3.0f, 3.0f } };
 
+    private static final Map<VectorSimilarityFunction, Function<Float, Float>> VECTOR_SIMILARITY_TO_SCORE = ImmutableMap.of(
+        VectorSimilarityFunction.EUCLIDEAN,
+        (similarity) -> 1 / (1 + similarity),
+        VectorSimilarityFunction.DOT_PRODUCT,
+        (similarity) -> (1 + similarity) / 2,
+        VectorSimilarityFunction.COSINE,
+        (similarity) -> (1 + similarity) / 2
+    );
```

Comment:
Are these functions correct? For cosine, is it not ```1/(1 + similarity)```?

Reply:
I took it from [Lucene implementation](https://github.com/apache/lucene/blob/branch_9_2/lucene/core/src/java/org/apache/lucene/index/VectorSimilarityFunction.java#L72-L74). Seems with current implementation IT tests are passing for cosine - https://github.com/opensearch-project/k-NN/blob/main/src/test/java/org/opensearch/knn/index/LuceneEngineIT.java#L56

Reply:
seems all tests are green